From 75fc6767a9b326c3eb13f5f8b7bf7dabe65a3979 Mon Sep 17 00:00:00 2001
From: Charlotte Van Petegem <charlotte.vanpetegem@ugent.be>
Date: Wed, 3 Aug 2022 14:11:03 +0200
Subject: [PATCH] fix: roi: make sure empty cashflows are skipped when
 determining first cashflow

Empty cashflows are added when the begin date of the report is before the first
transaction.
---
 hledger/Hledger/Cli/Commands/Roi.hs | 15 +++++++++++----
 hledger/test/roi.test               | 21 +++++++++++++++++++++
 2 files changed, 32 insertions(+), 4 deletions(-)

diff --git a/hledger/test/roi.test b/hledger/test/roi.test
index d360f06d0..11175efa5 100644
--- a/hledger/test/roi.test
+++ b/hledger/test/roi.test
@@ -345,3 +345,24 @@ hledger -f - roi --inv saving --pnl dividend
 +---++------------+------------++---------------+----------+-------------+---------++-------+-------+
 
 >>>=0
+
+# 14. Should support begin date before first transaction where first transaction has pnl
+hledger -f - roi --inv stocks --pnl expenses --value=then,€ -Y
+<<<
+P 2022-07-31 A € 1
+
+2022-08-01 Purchase
+    checking   € -101
+    stocks  100 A @ € 1
+    expenses   € 1
+
+P 2022-08-02 A € 2
+
+>>>
++---++------------+------------++---------------+----------+-------------+------++---------+--------+
+|   ||      Begin |        End || Value (begin) | Cashflow | Value (end) |  PnL ||     IRR |    TWR |
++===++============+============++===============+==========+=============+======++=========+========+
+| 1 || 2022-01-01 | 2022-12-31 ||             0 |    € 101 |       € 200 | € 99 || 410.31% | 98.02% |
++---++------------+------------++---------------+----------+-------------+------++---------+--------+
+
+>>>=0
-- 
2.25.1

